src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -140,31 +148,38 @@ const FiatDetailsScreen = ({ route, navigation }: Props) => {
         implicitBody[param.name] = param.value
       })
 
+      const fiatAccountSchema = quote.getFiatAccountSchema()
       const completeBody = {
         ...body,
         ...implicitBody,
       }
 
-      // await addNewFiatAccount(quote.getProviderBaseUrl(), fiatAccountSchema, completeBody)
-      //   .then((data) => {
-      //     // TODO Tracking here
-      //     dispatch(showMessage(t('fiatDetailsScreen.addFiatAccountSuccess')))
-      //   })
-      //   .catch((error) => {
-      //     // TODO Tracking here
-      //     if (error === FiatConnectError.ResourceExists) {
-      //       dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
-      //     } else {
-      //       dispatch(showError(t('fiatDetailsScreen.addFiatAccountFailed')))
-      //     }
-      //     Logger.error(TAG, `Error adding fiat account: ${error}`)
-      //   })
-
-      navigate(Screens.FiatConnectReview, {
-        flow,
-        normalizedQuote: quote,
-        fiatAccount: completeBody as FiatAccount,
+      const fiatConnectClient = await quote.getFiatConnectClient()
```

Comment:
*a wild kathy appeared* this might be something we can consider in a separate PR, but is it possible to abstract this kind of logic to a saga? this would better separate concerns, where this component can focus on displaying things and dispatch an action that kicks off async logic. there are some added benefits to this (e.g. easier to test because it is easier to test async logic in a saga, shorter UI file) happy to chat if i've not been clear here! :)

Reply:
Hmm, that would involve storing state in redux about, e.g., the most recent attempt to send fiat account information to a provider, right? Would the general strategy here look something like:
* Select the piece of state containing the add account response in the FiatDetailsScreen component
* While the add account response is null/empty, do nothing
* When the button is pressed, dispatch an action to add the account and save the response in state
* While the add account response is defined, check the result and either show an error or progress to the next page
Apologies if that's totally off :slightly_smiling_face:

Reply:
On activation we have been doing a small amount of our work in saga's, but mostly been leaving this kind of logic to helper functions outside of UI components. I agree that we should be doing more in saga's and since this is a fresh implementation it would be a good place to start! :) 

@jophish thats not quite the model I was thinking of for this type of an event since we don't *need* to store any data in state yet.

I think you can do all the logic you currently have with an Action, a watch* saga, and a saga function. All the navigation / conditionally showing errors that you currently have in this function can live in the saga.

For example take a look at account initialization:

* Action: https://github.com/valora-inc/wallet/blob/main/src/account/actions.ts#L317
* watch* saga: https://github.com/valora-inc/wallet/blob/main/src/account/saga.ts#L256
* saga: https://github.com/valora-inc/wallet/blob/main/src/account/saga.ts#L87

Together they watch for that INITIALIZE_ACCOUNT action in redux and call the initializeAccount() saga when its seen.

Reply:
The only question I have for @kathaypacific on this approach though is passing the `quote` object or `fiatConnectClient` (both of which are instantiated classes) through the dispatch action so that it can be used in the saga. Is it okay to do this? I know its generally best practice to stick with minimal data types in redux

Reply:
Oh @jeanregisser maybe you can comment on this approach since Kathy is on vacation now :)

Reply:
good idea @kathaypacific and thanks for explaining @jh2oman !